vagrant test environment not working

I tried to fire up the vagrant environment and run into a problem. When running the bootstrapping, the restart of the joiner nodes runs into a timeout. When looking into the logs of galera-db-2, I found the following error messages:

```
Feb 07 08:28:03 galera-db-2 mysqld[21769]: 2018-02-07  8:28:03 140223543498496 [ERROR] WSREP: Failed to read 'ready <addr>' from: wsrep_sst_xtrabackup-v2 --role 'joiner' --address '172.27.47.22...nt '21769'  ''
Feb 07 08:28:03 galera-db-2 mysqld[21769]: Read: '(null)'
Feb 07 08:28:03 galera-db-2 mysqld[21769]: 2018-02-07  8:28:03 140223543498496 [ERROR] WSREP: Process completed with error: wsrep_sst_xtrabackup-v2 --role 'joiner' --address '172.27.47.22' --da... (Broken pipe)
Feb 07 08:28:03 galera-db-2 mysqld[21769]: 2018-02-07  8:28:03 140223855065856 [ERROR] WSREP: Failed to prepare for 'xtrabackup-v2' SST. Unrecoverable.
Feb 07 08:28:03 galera-db-2 mysqld[21769]: 2018-02-07  8:28:03 140223855065856 [ERROR] Aborting
```

Seems to be a communication problem between the joiner node and the "master" or something?

Thank you!

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -6,6 +6,11 @@ CLUSTER_SIZE = 3
 Vagrant.configure("2") do |config|
   config.vm.box = "centos/7"
 
+  # default to libvirt provider if possible
+  config.vm.provider "libvirt"
+  # fall back to virtualbox if needed
+  config.vm.provider "virtualbox"
+
   (1..CLUSTER_SIZE).each do |box_id|
     config.vm.define "galera-db-#{box_id}" do |box|
       box.vm.hostname = "galera-db-#{box_id}"

--- a/roles/galera_packages/tasks/main.yml
+++ b/roles/galera_packages/tasks/main.yml
@@ -6,11 +6,19 @@
     - MariaDB-server
     - galera
 
-- name: Install Percona tool packages
+- name: Install percona-xtrabackup packages
   yum: name={{ item }} state=present
   with_items:
     - percona-xtrabackup
     - socat
+  when: "mariadb_release | version_compare('10.2', '<')"
+
+- name: Install percona-xtrabackup-24 packages
+  yum: name={{ item }} state=present
+  with_items:
+    - percona-xtrabackup-24
+    - socat
+  when: "mariadb_release | version_compare('10.2', '>=')"
 
 - name: Install MySQL-python
   yum: name=MySQL-python state=present

--- a/roles/galera_repos/defaults/main.yml
+++ b/roles/galera_repos/defaults/main.yml
@@ -1,4 +1,2 @@
 ---
-mariadb_release: 10.1
-percona_release_version: 0.1
-percona_release_package_version: "{{ percona_release_version }}-3"
+mariadb_release: 10.2

--- a/roles/galera_repos/tasks/main.yml
+++ b/roles/galera_repos/tasks/main.yml
@@ -2,9 +2,3 @@
 - name: Add MariaDB repo
   template: src=yum.repos.d/mariadb.repo.j2 dest=/etc/yum.repos.d/mariadb.repo
     owner=root group=root mode=0644
-
-- name: Add EPEL repo
-  yum:
-    name: epel-release
-    update_cache: yes
-    state: present

--- a/roles/galera_repos/templates/yum.repos.d/mariadb.repo.j2
+++ b/roles/galera_repos/templates/yum.repos.d/mariadb.repo.j2
@@ -1,7 +1,13 @@
-# MariaDB {{ mariadb_release }} CentOS repository list
-# http://mariadb.org/mariadb/repositories/
-[mariadb]
-name = MariaDB
-baseurl = http://yum.mariadb.org/{{ mariadb_release }}/centos7-amd64
-gpgkey=https://yum.mariadb.org/RPM-GPG-KEY-MariaDB
-gpgcheck=1
+[mariadb-main]
+name = MariaDB Server
+baseurl = https://downloads.mariadb.com/MariaDB/mariadb-{{ mariadb_release }}/yum/rhel/$releasever/$basearch
+gpgkey = https://downloads.mariadb.com/MariaDB/RPM-GPG-KEY-MariaDB
+gpgcheck = 1
+enabled = 1
+
+[mariadb-tools]
+name = MariaDB Tools
+baseurl = https://downloads.mariadb.com/Tools/rhel/$releasever/$basearch
+gpgkey = https://downloads.mariadb.com/Tools/MariaDB-Enterprise-GPG-KEY
+gpgcheck = 1
+enabled = 1

